rclcpp/include/rclcpp/mapped_ring_buffer.hpp

```diff
@@ -108,24 +109,25 @@ class MappedRingBuffer : public MappedRingBufferBase
     auto it = get_iterator_of_key(key);
     value = nullptr;
     if (it != elements_.end() && it->in_use) {
-      T * raw_ptr;
-      ElemDeleter * deleter;
       if (it->unique_value) {
-        raw_ptr = it->unique_value.get();
-        deleter = &it->unique_value.get_deleter();
-      } else {
-        raw_ptr = const_cast<T *>(it->shared_value.get());
-        deleter = std::get_deleter<ElemDeleter, const T>(it->shared_value);
+        ElemDeleter deleter = it->unique_value.get_deleter();
+        auto ptr = ElemAllocTraits::allocate(*allocator_.get(), 1);
+        ElemAllocTraits::construct(*allocator_.get(), ptr, *it->unique_value);
+        value = ElemUniquePtr(ptr, deleter);
+        return;
       }
-      if (raw_ptr) {
+      if (it->shared_value) {
+        ElemDeleter * deleter = std::get_deleter<ElemDeleter, const T>(it->shared_value);
         auto ptr = ElemAllocTraits::allocate(*allocator_.get(), 1);
-        ElemAllocTraits::construct(*allocator_.get(), ptr, *raw_ptr);
+        ElemAllocTraits::construct(*allocator_.get(), ptr, *it->shared_value);
         if (deleter) {
           value = ElemUniquePtr(ptr, *deleter);
         } else {
           value = ElemUniquePtr(ptr);
         }
+        return;
       }
+      throw std::runtime_error("Unexpected empty MappedRingBuffer element.");
```

Comment:
It's a bit weird that here you throw at the end of the function, but in a similar case below you throw from the else case, use if-else for the shared value case, and do not return from those. I prefer the style in the case below to the one here, but either way it would be better to be consistent.